src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -173,8 +173,9 @@ public final Connection getConnection(final long hardTimeout) throws SQLExceptio
                timeout = hardTimeout - clockSource.elapsedMillis(startTime);
             }
             else {
-               metricsTracker.recordBorrowStats(poolEntry, startTime);
-               return poolEntry.createProxyConnection(leakTask.start(poolEntry), now);
+               poolEntry.lastBorrowed = now;
+               metricsTracker.recordBorrowStats(poolEntry, startTime, now);
+               return poolEntry.createProxyConnection(leakTask.schedule(poolEntry, houseKeepingExecutorService), now);
```

Comment:
``lastBorrowed`` is only used for metrics.  By allowing the ``metricsTracker`` to set (or not set) ``lastBorrowed`` we completely avoid the assignment when metrics are disabled.  The [NopMetricsTrackerDelegate](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/PoolBase.java#L642) does nothing, whereas the [MetricsTrackerDelegate](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/PoolBase.java#L614) does set ``lastBorrowed``.  I am fine with lines 177-178, but 176 should be deleted and put back in the ``MetricsTrackerDelegate``.